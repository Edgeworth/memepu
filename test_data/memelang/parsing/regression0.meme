fn print_cstr(s: *u8) {
  while *s != 0 {
    printf("%c", *(s++));
  }
}
