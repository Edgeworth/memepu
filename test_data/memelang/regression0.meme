// status: ok
// input: ''
// output: ''
// returns: 0
fn print_cstr(s: *u8) {
  while *s != 0 {
    printf("%c", *(s++));
  }
}
fn main() i8 { ret 0; }
