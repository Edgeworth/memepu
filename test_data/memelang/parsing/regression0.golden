Module
 File(regression0.meme)
  Fn(print_cstr)
   FnSig
    Typename(print_cstr)
    VarDecl(s)
     Type(Ref(u8))
      Ref(u8)
      Qualifier(const: 0 ptr: 1, array: 0)
   StmtBlk
    While
     Op(NEQ)
      Op(UNARY_DEREF)
       Ref(s)
      IntLit(0)
     StmtBlk
      Op(FN_CALL)
       Ref(printf)
       FnCallArgs
        StrLit(%c)
        Op(UNARY_DEREF)
         Op(POSTFIX_INC)
          Ref(s)
  Fn(main)
   FnSig
    Typename(main)
    Type(Ref(i8))
     Ref(i8)
   StmtBlk
    Ret
     IntLit(0)
